File: app/src/main/java/com/hjc/reader/adapter/ImageAdapter.java
package com.hjc.reader.adapter;

import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.reader.R;
import com.hjc.reader.databinding.ItemImageBinding;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class ImageAdapter extends BaseQuickAdapter<String, BaseViewHolder> {

    public ImageAdapter(@Nullable List<String> data) {
        super(R.layout.item_image, data);
    }

    @Override
    protected void onItemViewHolderCreated(@NotNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NotNull BaseViewHolder helper, String item) {
        if (item == null) {
            return;
        }

        ItemImageBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setImageUrl(item);
        }
    }
}

File: module-gank/src/main/java/com/hjc/learn/gank/adapter/ImageAdapter.kt
package com.hjc.learn.gank.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.learn.gank.R
import com.hjc.learn.gank.databinding.GankItemImageBinding

/**
 * @Author: HJC
 * @Date: 2021/2/23 16:59
 * @Description: 图片adapter
 */
class ImageAdapter(data: MutableList<String?>?) :
    BaseQuickAdapter<String?, BaseViewHolder>(R.layout.gank_item_image, data) {

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: String?) {
        if (item == null){
            return
        }
        val binding = DataBindingUtil.getBinding<GankItemImageBinding>(holder.itemView)
        binding?.imageUrl = item
    }
}